src/sentry/tagstore/snuba/backend.py

```diff
@@ -380,9 +380,31 @@ def get_groups_user_counts(self, project_id, group_ids, environment_id):
                              referrer='tagstore.get_groups_user_counts')
         return defaultdict(int, {k: v for k, v in result.items() if v})
 
-    def get_group_event_ids(self, project_id, group_id, environment_id, tags):
-        # This method is not implemented since the `event.id` column doesn't
-        # exist in Snuba.
+    def get_group_event_filter(self, project_id, group_id, environment_id, tags):
+        start, end = self.get_time_range()
+        filters = {
+            'project_id': [project_id],
+            'environment': [environment_id],
```

Comment:
This actually feels wrong, as tagstore methods should allow for `environment_id = None` to mean "all environments." But it seems to be all over the Snuba tagstore backend, so I'm going to do a follow on PR with fixes/tests.

Reply:
Yeah so the current situation with the util is basically.
`'environment': [None]` gets stripped, so that there are no environment filters at all
`'environment': []` Same thing, no environment filters, all envs returned.
`'environment': ['']`: Gets translated into `environment IS NULL` which gives us just the events with the default environment (because we were saving the default environment as Null in snuba)

Reply:
Ahhh OK. I guess my brain is still so deep in the other implementations that it just read like a bug. Makes sense, I guess this would have been very noticeable in comparisons.

Reply:
Yeah its a little gross and I hate mixing all those falsy things like None, empty strings, and empty lists, in python. But it also seemed a little neater than having to conditionally add the environment filter `if environment_id is not None` in all these methods.